### PR TITLE
[SYCLomatic] Fix bug that CMakeLists.txt not migrated on Windows and add one new error code for cmake script migration.

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/Error.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ void ShowStatus(int Status, std::string Message) {
     StatusString = "Migration process completed";
     break;
   case MigrationNoCodeChangeHappen:
-    StatusString = "Migration not necessary; no CUDA code detected";
+    StatusString = "No CUDA source code detected";
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Error.cpp]: I'd suggest add new error code for cmake script migration.

File: clang/lib/DPCT/MigrateCmakeScript.cpp [hunk 2]
```diff
@@ -234,7 +234,7 @@ void collectCmakeScripts(const clang::tooling::UnifiedPath &InRoot,
     if (Iter->type() == fs::file_type::regular_file) {
       llvm::StringRef Name =
           llvm::sys::path::filename(FilePath.getCanonicalPath());
-      if (Name == "CMakeLists.txt" || Name.ends_with(".cmake")) {
+      if (Name.lower() == "cmakelists.txt" || Name.ends_with(".cmake")) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrateCmakeScript.cpp]: let's distinguish for windows and Linux,  as windows filename is capital insensitive.

Comment (zhimingwang36) [reply clang/lib/DPCT/MigrateCmakeScript.cpp]: same for ".cmake" compare.

File: clang/lib/DPCT/Error.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/Error.cpp]: Okay

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**